### DISCUSSION TITLE
Support variables in Filter by name transformation

### DISCUSSION THREAD

**Discussion by ko5win** [grafana/grafana#50451] [Dashboards, Panels & Visualizations]
Reference: 

[this issue](https://github.com/grafana/grafana/issues/42267) 
[this comment](https://github.com/grafana/grafana/issues/25469#issuecomment-1126696130)

It seems that [a change](https://github.com/grafana/grafana/pull/44486) has been made to support using template variables from within Transformers. However, the `Filter by name` transformation regex does not recognize the variable syntax.

It would be very useful if the regex filter can dynamically react to user defined variables, making dashboards to be more interactive in general.

**Comment by Kirchen99** [comment]
In the pull request https://github.com/grafana/grafana/pull/44486 only `calcultedFields` is enabled to support variables. We need this feature for the transformation "Filter by Name" as well.

Also there are other Grafana users need this feature: 
- https://github.com/grafana/grafana/issues/25469#issuecomment-1126696130
- https://github.com/grafana/grafana/discussions/50451
- https://github.com/grafana/grafana/issues/42267
- and I

**Comment by Kirchen99** [comment]
Please have a look at this PR https://github.com/grafana/grafana/pull/51804, maybe it solves the problem.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
